Automattic/jetpack
# JS Components: Convert admin section (AdminSection, AdminSectionHero) components to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `AdminSection` and `AdminSectionHero` components to TypeScript.
* It also converts the `AdminSection` stories to TS

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the admin section stories work as expected
* Check the consumers of `AdminSection` and `AdminSectionHero` and related components and confirm that they work as expected
* Check for any regression

<img width="1284" alt="Screenshot 2022-08-03 at 12 56 24 PM" src="https://user-images.githubusercontent.com/18226415/182549297-23345b77-79dd-4f90-8097-affab26819b5.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202276314263995
